pkg/controllers/customeradmin/group_mapping_test.go

```diff
@@ -309,6 +309,38 @@ func TestFromMSGraphGroup(t *testing.T) {
 				Users: []string{"user@home.com"},
 			},
 		},
+		{
+			name:          "domain case can match",
+			kubeGroupName: osaCustomerAdmins,
+			want1:         true,
+			msGroupMembers: []graphrbac.User{
+				{
+					Mail: to.StringPtr("user@Home.com"),
+				},
+			},
+			want: &v1.Group{
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name: osaCustomerAdmins,
+				},
+				Users: []string{"user@home.com"},
+			},
+		},
+		{
+			name:          "user case can't match",
```

Comment:
I don't really understand this test case, how is it different than e.g. the one on L234?

Reply:
previously in some cases we were incorrectly doing case insensitive match on the whole e-mail address.  In fact the domain part is case insensitive and the username part is case sensitive.  The first additional test case checks that we correctly coalesce Home.com -> home.com; the second checks that we do not incorrectly coalesce User@ to user@.